### ISSUE TITLE
ERROR: [057]: no statement executing

### ISSUE THREAD

**Issue by AnzeJ** [pgbackrest/pgbackrest#419]
I'm getting the following error trying to backup PostgreSQL 9.6.5 and 10 server running inside Debian Stretch Docker container from remote pgBackRest 1.24 server:

```
2017-10-05 20:43:44.233 P00   INFO: full backup size = 47.8MB
2017-10-05 20:43:44.233 P00   INFO: execute non-exclusive pg_stop_backup() and wait for all WAL segments to archive
2017-10-05 20:43:44.234 P00  DEBUG:     Db->executeSqlRow(): strSql = select to_char(clock_timestamp(), 'YYYY-MM-DD HH24:MI:SS.US TZ'), pg_walfile_name(lsn), lsn::text, labelfile, case when length(trim(both '\t\n ' from spcmapfile)) = 0 then null else spcmapfile end as spcmapfile from pg_stop_backup(false, false) as lsn
2017-10-05 20:43:44.234 P00  DEBUG:     Db::executeSql(): bIgnoreError = <false>, bResult = <true>, strSql = select to_char(clock_timestamp(), 'YYYY-MM-DD HH24:MI:SS.US TZ'), pg_walfile_name(lsn), lsn::text, labelfile, case when length(trim(both '\t\n ' from spcmapfile)) = 0 then null else spcmapfile end as spcmapfile from pg_stop_backup(false, false) as lsn
2017-10-05 20:43:44.238 P00  ERROR: [057]: raised on '*************** remote' host: no statement executing:
                                        select to_char(clock_timestamp(), 'YYYY-MM-DD HH24:MI:SS.US TZ'), pg_walfile_name(lsn), lsn::text, labelfile, case when length(trim(both '\t\n ' from spcmapfile)) = 0 then null else spcmapfile end as spcmapfile from pg_stop_backup(false, false) as lsn
 at /usr/share/perl5/pgBackRest/Protocol/Base/Master.pm line 173.
                                        pgBackRest::Protocol::Base::Master::outputRead(undef, 1, undef, undef) called at /usr/share/perl5/pgBackRest/Protocol/Base/Master.pm line 255
                                        pgBackRest::Protocol::Base::Master::cmdExecute('pgBackRest::Protocol::Remote::Master=HASH(0x18b9c58)', 'dbExecSql', 'ARRAY(0x246a5b8)', 1) called at /usr/share/perl5/pgBackRest/Db.pm line 252
                                        pgBackRest::Db::executeSql(undef, 'select to_char(clock_timestamp(), \'YYYY-MM-DD HH24:MI:SS.US ...') called at /usr/share/perl5/pgBackRest/Db.pm line 362
                                        pgBackRest::Db::executeSqlRow(undef, 'select to_char(clock_timestamp(), \'YYYY-MM-DD HH24:MI:SS.US ...') called at /usr/share/perl5/pgBackRest/Db.pm line 703
                                        pgBackRest::Db::backupStop('pgBackRest::Db=HASH(0x18b60e8)') called at /usr/share/perl5/pgBackRest/Backup/Backup.pm line 832
                                        pgBackRest::Backup::Backup::process('pgBackRest::Backup::Backup=HASH(0x12ce680)') called at /usr/bin/pgbackrest line 252
                                        eval {...} called at /usr/bin/pgbackrest line 282

2017-10-05 20:43:44.242 P00  DEBUG:     Common::Exit::exitSafe(): iExitCode = [undef], oException = [object], strSignal = [undef]

```

PostgreSQL server log:

```
2017-10-05 20:43:39.904 CEST [1] LOG:  server process (PID 58) exited with exit code 255
2017-10-05 20:43:39.904 CEST [1] LOG:  terminating any other active server processes
2017-10-05 20:43:39.904 CEST [36] WARNING:  terminating connection because of crash of another server process
2017-10-05 20:43:39.904 CEST [36] DETAIL:  The postmaster has commanded this server process to roll back the current transaction and exit, because another server process exited abnormally and possibly corrupted shared memory.
2017-10-05 20:43:39.904 CEST [36] HINT:  In a moment you should be able to reconnect to the database and repeat your command.
2017-10-05 20:43:39.906 CEST [47] WARNING:  terminating connection because of crash of another server process
2017-10-05 20:43:39.906 CEST [47] DETAIL:  The postmaster has commanded this server process to roll back the current transaction and exit, because another server process exited abnormally and possibly corrupted shared memory.
2017-10-05 20:43:39.906 CEST [47] HINT:  In a moment you should be able to reconnect to the database and repeat your command.
2017-10-05 20:43:39.906 CEST [37] WARNING:  terminating connection because of crash of another server process
2017-10-05 20:43:39.906 CEST [37] DETAIL:  The postmaster has commanded this server process to roll back the current transaction and exit, because another server process exited abnormally and possibly corrupted shared memory.
2017-10-05 20:43:39.906 CEST [37] HINT:  In a moment you should be able to reconnect to the database and repeat your command.
2017-10-05 20:43:39.908 CEST [34] WARNING:  terminating connection because of crash of another server process
2017-10-05 20:43:39.908 CEST [34] DETAIL:  The postmaster has commanded this server process to roll back the current transaction and exit, because another server process exited abnormally and possibly corrupted shared memory.
2017-10-05 20:43:39.908 CEST [34] HINT:  In a moment you should be able to reconnect to the database and repeat your command.
2017-10-05 20:43:39.908 CEST [35] WARNING:  terminating connection because of crash of another server process
2017-10-05 20:43:39.908 CEST [35] DETAIL:  The postmaster has commanded this server process to roll back the current transaction and exit, because another server process exited abnormally and possibly corrupted shared memory.
2017-10-05 20:43:39.908 CEST [35] HINT:  In a moment you should be able to reconnect to the database and repeat your command.
2017-10-05 20:43:39.909 CEST [33] WARNING:  terminating connection because of crash of another server process
2017-10-05 20:43:39.909 CEST [33] DETAIL:  The postmaster has commanded this server process to roll back the current transaction and exit, because another server process exited abnormally and possibly corrupted shared memory.
2017-10-05 20:43:39.909 CEST [33] HINT:  In a moment you should be able to reconnect to the database and repeat your command.
2017-10-05 20:43:39.911 CEST [32] WARNING:  terminating connection because of crash of another server process
2017-10-05 20:43:39.911 CEST [32] DETAIL:  The postmaster has commanded this server process to roll back the current transaction and exit, because another server process exited abnormally and possibly corrupted shared memory.
2017-10-05 20:43:39.911 CEST [32] HINT:  In a moment you should be able to reconnect to the database and repeat your command.
2017-10-05 20:43:39.912 CEST [31] WARNING:  terminating connection because of crash of another server process
2017-10-05 20:43:39.912 CEST [31] DETAIL:  The postmaster has commanded this server process to roll back the current transaction and exit, because another server process exited abnormally and possibly corrupted shared memory.
2017-10-05 20:43:39.912 CEST [31] HINT:  In a moment you should be able to reconnect to the database and repeat your command.
2017-10-05 20:43:39.913 CEST [27] WARNING:  terminating connection because of crash of another server process
2017-10-05 20:43:39.913 CEST [27] DETAIL:  The postmaster has commanded this server process to roll back the current transaction and exit, because another server process exited abnormally and possibly corrupted shared memory.
2017-10-05 20:43:39.913 CEST [27] HINT:  In a moment you should be able to reconnect to the database and repeat your command.
2017-10-05 20:43:39.916 CEST [1] LOG:  archiver process (PID 28) exited with exit code 1
2017-10-05 20:43:39.917 CEST [1] LOG:  all server processes terminated; reinitializing
2017-10-05 20:43:39.929 CEST [61] LOG:  database system was interrupted; last known up at 2017-10-05 20:43:19 CEST
2017-10-05 20:43:39.961 CEST [61] LOG:  database system was not properly shut down; automatic recovery in progress
2017-10-05 20:43:39.962 CEST [61] LOG:  redo starts at 0/8000028
2017-10-05 20:43:39.962 CEST [61] LOG:  invalid record length at 0/8000140: wanted 24, got 0
2017-10-05 20:43:39.962 CEST [61] LOG:  redo done at 0/8000108
2017-10-05 20:43:39.970 CEST [1] LOG:  database system is ready to accept connections
2017-10-05 20:43:39.991 CEST [1] LOG:  server process (PID 44) exited with exit code 255
2017-10-05 20:43:39.991 CEST [1] LOG:  terminating any other active server processes
2017-10-05 20:43:39.992 CEST [65] WARNING:  terminating connection because of crash of another server process
2017-10-05 20:43:39.992 CEST [65] DETAIL:  The postmaster has commanded this server process to roll back the current transaction and exit, because another server process exited abnormally and possibly corrupted shared memory.
2017-10-05 20:43:39.992 CEST [65] HINT:  In a moment you should be able to reconnect to the database and repeat your command.
2017-10-05 20:43:39.995 CEST [1] LOG:  archiver process (PID 66) exited with exit code 1
2017-10-05 20:43:39.997 CEST [1] LOG:  all server processes terminated; reinitializing
2017-10-05 20:43:40.010 CEST [69] LOG:  database system was interrupted; last known up at 2017-10-05 20:43:39 CEST
2017-10-05 20:43:40.034 CEST [69] LOG:  database system was not properly shut down; automatic recovery in progress
2017-10-05 20:43:40.035 CEST [69] LOG:  invalid record length at 0/80001B0: wanted 24, got 0
2017-10-05 20:43:40.035 CEST [69] LOG:  redo is not required
2017-10-05 20:43:40.042 CEST [1] LOG:  database system is ready to accept connections
```

pgBackRest configuration on backup server:

```
[*******]
db1-host=***************
db1-path=/var/lib/postgresql/data
db1-user=postgres

[global]
repo-path=/var/lib/pgbackrest
retention-full=2
start-fast=y
log-level-file=debug
```

pgBackRest configuration on PostgreSQL server:

```
[*******]
db-path=/var/lib/postgresql/data

[global]
backup-host=backup
backup-user=backrest
```

Thank you and let me know if there is any additional information I can provide.

Anže

**Comment by dwsteele** [comment]
Interesting - it looks like a backend is crashing, but I don't know if it's the one that pgBackRest is connected to.  It looks like not, based on the error, but I can't be sure.

We have fixed a few bugs in 9.6/10 related to pg_stop_backup() crashing the backend but those were usually caused by pg_stop_backup() being run multiple times, mixing exclusive/non-exclusive, etc.  None of the bugs affected pgBackRest.

Clearly there are a number of backends running -- does this issue only happen when running a backup?

**Comment by AnzeJ** [comment]
After some troubleshooting, I found out that sshd daemon running inside container is causing PostgreSQL to crash. Someone already opened an issue about it on [docker-library/postgres](https://github.com/docker-library/postgres) repository's page ([#314](https://github.com/docker-library/postgres/issues/314)). After I replaced OpenSSH server with Dropbear, I could backup database without PostgreSQL crashing. This issue can be closed as it's obviously not related to pgBackRest.
